reconcile/openshift_saas_deploy.py

```diff
@@ -99,6 +99,7 @@ def run(
     env_name: Optional[str] = None,
     trigger_integration: Optional[str] = None,
     trigger_reason: Optional[str] = None,
+    all_saas_files: Optional[list[SaasFile]] = None,
```

Comment:
shouldn't there be a `if` on line 108 below?

```
if all_saas_file is None:
  all_saas_files = get_saas_files()

Reply:
Also, could `get_saas_files(saas_file_name, env_name)` accept `all_saas_files` as a parameter?

Reply:
did the first in https://github.com/app-sre/qontract-reconcile/pull/3839

will avoid the second today :pray: